### ISSUE TITLE
PyCUDA context stack not empty

### ISSUE THREAD

**Issue by gschramm** [jyhmiinlin/pynufft#40]
When I run a pynufft with pycuda backend, at the end of a program, I get the
error that PyCUDA Error: The context stack was not empty upon module clean up.

Is that expected?

I am running 2022.22.3

**Comment by jyhmiinlin** [comment]
I cannot replicate the error. No such error in my current system. I am using 
Ubuntu 22.4
python3-pycuda/jammy,now 2021.1
NVIDIA-SMI 510.108.03   
Driver Version: 510.108.03   
CUDA Version: 11.6 
PyNUFFT 2022.2.3

A Google search suggested possible Torch - pycuda interactions?

**Comment by gschramm** [comment]
Ok, I will do some further testing. 
Could it be related to the fact that I installed the packge from "conda-forge" and not from pypi?

**Comment by jyhmiinlin** [comment]
It might be. I have not tested conda-forge.

**Comment by jake106** [comment]
I am having the same issue using:

python 3.8.7
pycuda 2022.1
NVIDIA-SMI 460.32.03
Driver Version: 460.32.03
CUDA Version: 11.2
PyNUFFT 2022.2.3 & 2020.2.7

All installed with pypi

**Comment by jyhmiinlin** [comment]
This issue happended when mixing mismatched versions of python, nvidia driver, and cuda. There is no clean solution to solve this issue. Current suggestion is to use pyopencl if this PyCUDA error persists. 

Google searching shows that it can be related to:

1. mismatched boost-lib version (https://bugs.archlinux.org/task/47605). Upgrading boost-lib to suitable version could be helpful.  
2. pytorch + tensorrt (https://forums.developer.nvidia.com/t/why-i-add-context-pop-its-also-report-pycuda-error-the-context-stack-was-not-empty-upon-module-cleanup/232993). I tested this issue but I got another one (pycuda._driver.LogicError: cuDeviceGet failed: initialization error).
